app/helpers/samples_helper.rb

```diff
@@ -100,38 +99,61 @@ def samples_output_data(samples)
     final_result
   end
 
+  def filter_samples(samples, query)
+    if query == 'WAITING'
+      samples = samples.joins("LEFT OUTER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where("samples.status = ?  or pipeline_runs.job_status is NULL", 'created')
+    elsif query == 'FAILED'
+      db_query = [query, '1.Host Filtering-FAILED', '2.GSNAPL/RAPSEARCH alignment-FAILED', '3.Post Processing-FAILED', 'ERROR']
+      samples = samples.joins("INNER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where(status: 'checked').where("pipeline_runs.id in (select max(id) from pipeline_runs group by sample_id)").where("pipeline_runs.job_status IN (?)", db_query)
+    elsif query == 'UPLOADING'
+      db_query = [query, '1.Host Filtering-STARTED', '1.Host Filtering-CHECKED', '1.Host Filtering-LOADED',
```

Comment:
this is problematic because there are also status like RUNNING and RUNNABLE. 
To make it simple you can just do 

where("pipeline_runs.job_status NOT IN (?) and pipeline_runs.finalized != 1",  ['CHECKED',  'FAILED'])